src/libraries/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/HashProviderDispenser.Unix.cs

```diff
@@ -25,6 +25,43 @@ internal static HashProvider CreateMacProvider(string hashAlgorithmId, ReadOnlyS
 
         internal static class OneShotHashProvider
         {
+            public static unsafe int MacData(
+                string hashAlgorithmId,
+                ReadOnlySpan<byte> key,
+                ReadOnlySpan<byte> source,
+                Span<byte> destination)
+            {
+                IntPtr evpType = Interop.Crypto.HashAlgorithmToEvp(hashAlgorithmId);
+                Debug.Assert(evpType != IntPtr.Zero);
+
+                int hashSize = Interop.Crypto.EvpMdSize(evpType);
+
+                if (hashSize <= 0 || destination.Length < hashSize)
+                {
+                    Debug.Fail("Destination length or hash size not valid.");
+                    throw new CryptographicException();
+                }
+
+                int length = destination.Length;
+
+                fixed (byte* pKey = key)
+                fixed (byte* pSource = source)
+                fixed (byte* pDestination = destination)
+                {
+                    Check(Interop.Crypto.HmacOneShot(
```

Comment:
I know that this is following the style of the other hashing, but that was done way early on in our cross-platform stuff, where we were really just shimming calls via the libraries still known as "the shims".

If there's not expected to be any calls to OneShot that do anything other than throw, then I think it'd make things looks cleaner to move all the pinning and such to an `internal static int [Interop.Crypto.]HmacOneShot(ROS key, ROS source, Span destination)` which does all the gymnastics here and calls a private DllImported method.